tests/chainer_tests/functions_tests/math_tests/test_sparse_matmul.py

```diff
@@ -0,0 +1,293 @@
+        self.forward_answer = self._matmul(self.a, self.b)
+
+    def _set_shape(self, shape, trans):
+        if trans:
+            shape = [shape[1], shape[0]]
+        if self.nbatch > 0:
+            shape = [self.nbatch, shape[0], shape[1]]
+        return shape
+
+    def _matmul(self, a, b):
+        if self.transa:
+            a = a.swapaxes(-1, -2)
+        if self.transb:
+            b = b.swapaxes(-1, -2)
+        return numpy.matmul(a, b)
+
+    #
+    # SPDN: sparse A * dense B
+    #
+    def check_SPDN_forward(self, a_data, b_data, atol=1e-4, rtol=1e-5):
+        sp_a = utils.sparse_dense2coo(a_data, use_variable=True)
+        b = chainer.Variable(b_data)
+        c = F.sparse_matmul(sp_a, b, transa=self.transa, transb=self.transb)
+        testing.assert_allclose(self.forward_answer, c.data, atol, rtol)
+
+    def test_SPDN_sparse_matmul_forward_cpu(self):
+        if not _scipy_available:
+            return
+        if self.a_dtype == numpy.float16 or self.b_dtype == numpy.float16:
+            self.check_SPDN_forward(self.a, self.b, atol=1e-3, rtol=1e-3)
+        else:
+            self.check_SPDN_forward(self.a, self.b)
+
+    @attr.gpu
+    def test_SPDN_sparse_matmul_forward_gpu(self):
+        a = cuda.to_gpu(self.a)
+        b = cuda.to_gpu(self.b)
+        if self.a_dtype == numpy.float16 or self.b_dtype == numpy.float16:
+            self.check_SPDN_forward(a, b, atol=1e-3, rtol=1e-3)
+        else:
+            self.check_SPDN_forward(a, b)
+
+    def check_SPDN_backward(self, a_data, b_data, c_grad, atol, rtol):
+        sp_a = utils.sparse_dense2coo(a_data)
+        func = F.math.sparse_matmul.SparseMatMul(
+            sp_a.row, sp_a.col, sp_a.shape,
+            transa=self.transa, transb=self.transb, transc=False)
+
+        def op(a, b):
+            return func.apply((a, b))[0]
+        gradient_check.check_backward(
+            op, (sp_a.data, b_data), c_grad, atol=atol, rtol=rtol,
+            dtype=numpy.float32)
+
+    def test_SPDN_sparse_matmul_backward_cpu(self):
+        if not _scipy_available:
+            return
+        self.check_SPDN_backward(
+            self.a, self.b, self.gc, atol=1e-2, rtol=1e-2)
+
+    @attr.gpu
+    def test_SPDN_sparse_matmul_backward_gpu(self):
+        self.check_SPDN_backward(
+            cuda.to_gpu(self.a), cuda.to_gpu(self.b),
+            cuda.to_gpu(self.gc), atol=1e-2, rtol=1e-2)
+
+    def check_SPDN_double_backward(
+            self, a_data, b_data, c_grad, a_grad_grad, b_grad_grad,
+            atol, rtol):
+        sp_a = utils.sparse_dense2coo(a_data)
+        sp_gga = utils.sparse_dense2coo(a_grad_grad)
+        func = F.math.sparse_matmul.SparseMatMul(
+            sp_a.row, sp_a.col, sp_a.shape,
+            transa=self.transa, transb=self.transb, transc=False)
+
+        def op(a, b):
+            return func.apply((a, b))[0]
+        gradient_check.check_double_backward(
+            op, (sp_a.data, b_data), c_grad, (sp_gga.data, b_grad_grad),
```

Comment:
Can you test the function interface without using the `FunctionNode` class + `apply`?

Reply:
The reason why I used `apply` here is that one of inputs to `sparse_matmul` is not `Variable` and is `CooMatrix`.  The check routines like `check_double_backward` in `gradient_check.py` gives tuple of `Variable` to `func`. So, `sparse_matmul` can not be used as `func` for those check routines, AFAIK.

Is there any better way to work-around this without using `apply`?